WordPress/gutenberg
# Embed block: Fix previewable embed URLs failure when using the Facebook embed block

## Description


I noticed that when upgrading an embed block, we're passing some of the attributes defined on the block that is being upgraded. Some of them like `previewable` or `responsive` are passed but might not be overridden by the attributes extracted from the new block (the more suitable one) because it could happen that some of them are not defined in the embed variation definition.

Here is an example following the reproduction steps described in the issue https://github.com/WordPress/gutenberg/issues/35226:

When adding the Facebook embed, we have the following attributes defined by the Facebook embed variation ([reference](https://github.com/WordPress/gutenberg/blob/255234cee2d54dd7972f75ea0f717be93c67ba12/packages/block-library/src/embed/variations.js#L59-L72)):
- `providerNameSlug: 'facebook'`
- `previewable: false`
- `responsive: true`

Then when setting a previewable embed URL like a YouTube URL (example: https://www.youtube.com/watch?v=ssfHW5lwFZg), the attributes defined for YouTube embed block are ([reference](https://github.com/WordPress/gutenberg/blob/255234cee2d54dd7972f75ea0f717be93c67ba12/packages/block-library/src/embed/variations.js#L46-L57)):
- `providerNameSlug: 'youtube'`
- `responsive: true`
- `url: 'https://www.youtube.com/watch?v=ssfHW5lwFZg'`

This means that when the function `createUpgradedEmbedBlock` is called, the newly created block that will replace the original one will end up having the following attributes:
- `url: 'https://www.youtube.com/watch?v=ssfHW5lwFZg'`
- **`previewable: false`**
- `providerNameSlug: 'youtube'`
- `responsive: true`

**`createUpgradedEmbedBlock` code reference:**
https://github.com/WordPress/gutenberg/blob/255234cee2d54dd7972f75ea0f717be93c67ba12/packages/block-library/src/embed/util.js#L99
https://github.com/WordPress/gutenberg/blob/255234cee2d54dd7972f75ea0f717be93c67ba12/packages/block-library/src/embed/util.js#L117-L121

As you can see in this case, we transferred wrongly the `previewable: false` attribute, leading to a YouTube embed block that can't be previewed.

This PR updates the `createUpgradedEmbedBlock` function and now we are only selecting the attributes that we know that the user could have modified and make sense to transfer to the new block.

## How has this been tested?



1. Add a Facebook embed block.
1. Set a previewable embed URL (example: https://www.youtube.com/watch?v=ssfHW5lwFZg).
1. Click on the Embed button.
1. Observe that the block is replaced by the YouTube block and that the preview is displayed.

**NOTE:** If the Facebook embed block is not available in the inserter menu, you can use the following HTML code:
```

```

## Screenshots 
N/A

## Types of changes




Bug fix

## Checklist:
